src/CodeGen_Hexagon.cpp

```diff
@@ -2232,6 +2232,24 @@ void CodeGen_Hexagon::visit(const Call *op) {
       Value *Call = CallLLVMIntrinsic(F, Ops);
       value = convertValueType(Call, llvm_type_of(op->type));
       return;
+    } else if (op->name == Call::interleave_vectors) {
+      if (isDblVector(op->type, native_vector_bits()) &&
+                      op->args.size() == 2) {
+        Expr arg0 = op->args[0];
+        Expr arg1 = op->args[1];
+        if ((arg0.type() == arg1.type()) &&
+            op->type.element_of() == arg0.type().element_of()) {
+          internal_assert(op->type.lanes() == 2*arg0.type().lanes());
+          int scalar = -1 * op->type.bytes();
+          std::vector<Value *> Ops = {codegen(arg1), codegen(arg0), codegen(scalar)};
```

Comment:
This is better, thanks. But just out of curiosity, it looks like CallLLVMIntrinsic takes a non-const reference just so it can change some of the arguments prior to calling IRBuilder::CreateCall... could it just take a vector by value instead? Do callers actually use the modified arguments? The performance impact would probably be negligible, and I think it would be a saner method.